### ISSUE TITLE
Panic observed when reading from perf ring buffer

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#5671]
```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x2 addr=0x7f6581be65d9 pc=0x8482d7]

runtime stack:
runtime.throw(0x96b574, 0x2a)
        /usr/local/go/src/runtime/panic.go:616 +0x81
runtime.sigpanic()
        /usr/local/go/src/runtime/signal_unix.go:372 +0x28e

goroutine 6 [syscall]:
runtime.cgocall(0x8483a0, 0xc42041fd90, 0x89a780)
        /usr/local/go/src/runtime/cgocall.go:128 +0x64 fp=0xc42041fd60 sp=0xc42041fd28 pc=0x403364
github.com/cilium/cilium/pkg/bpf._Cfunc_perf_event_read(0x7f6574000b50, 0xc4203e41b0, 0xc420917fc8, 0x0)
        _cgo_gotypes.go:165 +0x4d fp=0xc42041fd90 sp=0xc42041fd60 pc=0x823d1d
github.com/cilium/cilium/pkg/bpf.(*PerfEvent).Read.func2(0xc4203e4160, 0x7f6574000b50, 0xc4203e41b0, 0xc420917fc8, 0x1)
        /go/src/github.com/cilium/cilium/pkg/bpf/perf.go:322 +0x174 fp=0xc42041fdd0 sp=0xc42041fd90 pc=0x828474
github.com/cilium/cilium/pkg/bpf.(*PerfEvent).Read(0xc4203e4160, 0xc42041fef8, 0xc42041fee8)
        /go/src/github.com/cilium/cilium/pkg/bpf/perf.go:322 +0xa9 fp=0xc42041fe20 sp=0xc42041fdd0 pc=0x826949
github.com/cilium/cilium/pkg/bpf.(*PerCpuEvents).ReadAll(0xc420502000, 0xc42041fef8, 0xc42041fee8, 0x0, 0x0)
        /go/src/github.com/cilium/cilium/pkg/bpf/perf.go:493 +0x9f fp=0xc42041fe60 sp=0xc42041fe20 pc=0x82749f
main.(*Monitor).perfEventReader(0xc42043ec00, 0x9bbcc0, 0xc4201be000, 0x40)
        /go/src/github.com/cilium/cilium/monitor/monitor.go:223 +0x499 fp=0xc42041ffc0 sp=0xc42041fe60 pc=0x846f09
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc42041ffc8 sp=0xc42041ffc0 pc=0x45bc61
created by main.(*Monitor).registerNewListener
        /go/src/github.com/cilium/cilium/monitor/monitor.go:135 +0x641
```

**Comment by joestringer** [comment]
Fixed in v1.3.

**Comment by tgraf** [comment]
The change in https://github.com/cilium/cilium/pull/5842 points out an obvious potential buffer overflow due to `char buf[256]` which can clearly explain the above segfault. Adding a fixes tag to that PR for this issue.
